ruby/generate-website-screenshot/index.rb

```diff
@@ -0,0 +1,41 @@
+require 'net/http'
+require 'addressable/uri'
+require 'json'
+require 'base64'
+require 'open-uri'
+
+def main(request, response)
+    website_url = nil
+    begin
+        payload = JSON.parse(request.payload)
+        website_url = payload["url"]
+    rescue Exception => e
+        return response.json({"message":"Payload is invalid."})
+    end
+
+    begin
+        uri = Addressable::URI.parse("https://shot.screenshotapi.net/screenshot")
+        api_key = request.variables['SCREENSHOT_API_KEY']
+        uri.query_values = {
+          'token'  => api_key,
+          'url' => website_url,
+          'output' => 'JSON',
+          'full_page': true,
+        }
+        uri = URI(uri)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = true
+        req =  Net::HTTP::Get.new(uri)
+        res = http.request(req)
+
+        screenshot_url = JSON.parse(res.body)['screenshot']
+        image =  URI.open(screenshot_url)
+        result = Base64.strict_encode64(image.read)
+
+        return response.json({"success":true,"screenshot": result})
```

Comment:
Please add additional error handling in case the API call fails. For example, executing:

```shell
$ curl -H "X-Internal-Challenge: secret-key" -H "Content-Type: application/json" -X POST http://localhost:3000/ -d '{
    "variables": {"SCREENSHOT_API_KEY": "SCREENSHOT_API_KEY"},
    "payload": "{\"url\":\"https://google.com\"}"
}'
```

returns an unhelpful error message:

```json
{"response":{"success":false,"message":"no implicit conversion of nil into String"},"stdout":""}
```